include/perf_watcher.h

```diff
@@ -0,0 +1,109 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0. This product includes software
+// developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present
+// Datadog, Inc.
+
+#pragma once
+
+#include <linux/perf_event.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+struct PerfWatcherOptions {
+  bool is_kernel;
+  bool is_freq;
+};
+
+typedef struct PerfWatcher {
+  const char *desc;
+  uint64_t sample_type;
+  int type;
+  unsigned long config;
+  union {
+    uint64_t sample_period;
+    uint64_t sample_frequency;
+  };
+  int profile_id;
+  // perf_event_open configs
+  struct PerfWatcherOptions options;
+  int count_id;
+  // tracepoint configuration
+  uint8_t reg;
+  uint8_t trace_off;
+  uint8_t trace_sz;
+  const char *tracepoint_name;
+  const char *tracepoint_group;
+  // Other configs
+  bool send_pid;
+  bool send_tid;
+} PerfWatcher;
+
+// The Datadog backend only understands pre-configured event types.  Those
+// types are defined here, and then referenced in the watcher
+// The last column is a dependent type which is always aggregated as a count
+// whenever the main type is aggregated.
+#define PROFILE_TYPE_TABLE(X)                                                  \
+  X(NOCOUNT, nocount, nocount, NOCOUNT)                                        \
+  X(TRACEPOINT, tracepoint, events, NOCOUNT)                                   \
+  X(CPU_NANOS, cpu - time, nanoseconds, CPU_SAMPLE)                            \
```

Comment:
Why change the name for `cpu-time`, won't this impact the backend ?

Reply:
Sloppy mistake--I try to impose little structure on the X-table, in terms of strings, and the linter had no way of knowing that the `-` wasn't an operator.  I'll just write string-type columns as string literals.

Reply:
Fixed in latest commit